apply-roles

```diff
@@ -32,10 +38,12 @@ create_custom_role() {
     roleName="${CUSTOM_ROLE_NAME}" \
     subscriptionIds="${prefixed_subscription_ids}"
 
-  echo "Waiting for role to be searchable..."
-  sleep 10
-
-  custom_role_definition_id=$(az role definition list --name "${CUSTOM_ROLE_NAME}" --query "[0].name" --output tsv)
+  # It can take a bit of time for the role to be available to use so query until it is available
+  while [[ -z "${custom_role_definition_id}" ]] && [[ attempts_to_retrieve_role -lt 5 ]]; do
```

Comment:
5 attempts was an arbitrary number, though during testing it sometimes took only 1 attempt but never more than 3 so 5 attempts (i.e. an extra 10-15 seconds) should be long enough to wait for.

Reply:
😃 👍 You're effectively allowing it to wait as long as ~25 seconds (depending on the number of retries) or less (if role creation finishes sooner than 25 seconds) which is a nice property.